modules/custom-status/rest/custom-status-endpoint.php

```diff
@@ -339,17 +365,43 @@ public static function handle_reorder_status( WP_REST_Request $request ) {
 
 	// Utility functions
 
-	private static function is_valid_user_ids( $user_ids ) {
-		foreach ( $user_ids as $user_id ) {
-			$user = get_user_by( 'id', $user_id );
-			if ( ! $user instanceof WP_User ) {
+	private static function is_valid_editorial_metadata_ids( $metadata_ids ) {
+		// If the array is empty, it's valid
+		if ( [] === $metadata_ids ) {
+			return true;
+		}
+
+		// ToDo: Optimize this to batch fetch metadata terms, and only include the ID field.
+		foreach ( $metadata_ids as $metadata_id ) {
+			$editorial_metadata = EditorialMetadata::get_editorial_metadata_term_by( 'id', $metadata_id );
+			if ( is_wp_error( $editorial_metadata ) || ! $editorial_metadata ) {
 				return false;
 			}
 		}
 
 		return true;
 	}
 
+	private static function is_valid_user_ids( $user_ids ) {
```

Comment:
Made the user lookup a lot faster by making it a simple batch SQL query.

Reply:
Cool solution!